### ISSUE TITLE
Server Lag After 1.52 Patch

### ISSUE THREAD

**Issue by LouDnl** [A3Wasteland/ArmA3_Wasteland.Altis#354]
Could be Bohemia causing this, could be a part of code in the mission.
Creampie and I have been testing multiple solutions but all without any succes.
Disabled addons, disabled mArma, disabled ASM, changed basic.cfg etc.
Don't really know where to look now.

ref: http://forums.a3wasteland.com/index.php?topic=2150.0

**Comment by LouDnl** [comment]
Also created a feedback ticket: http://feedback.arma3.com/view.php?id=26108

**Comment by dagg929** [comment]
Tried swapping to default mission, no better-- when server is under load and full, it just randomly cuts out for however many seconds. 
Server FPS is good, steady, I get on our server always 15+ fps until this suddenly occurs.

Upvoting topic on A3 Feedback. Surprised to not see as much noise on the A3 forums about this, seems to be hitting everybody, regardless of mod. (Exile and Wasteland at least.)

**Comment by LouDnl** [comment]
Yes I have had this on Exile too. I am getting almost zero response in the server hosting Skype group.
I have disabled ASM now to see if that's the cause.

**Comment by dagg929** [comment]
I'm wondering if the reason I'm seeing this on Exile is because it runs on the same box as my A3Wasteland. Might be occurring at the same time? Talked to a few Exile server admins and they are not getting this.

**Comment by LouDnl** [comment]
I'm running all my servers on separate boxes. My Exile doesn't have much players but i've seen this choke atleast once.
Do you run ASM on your servers?

**Comment by dagg929** [comment]
Nope! I did turn it on however to try and monitor these issues just yesterday, but we were getting them before. It isn't ASM

**Comment by dagg929** [comment]
I also stopped running HC as we get it with or without, just to make the problem more simple.

**Comment by LouDnl** [comment]
Well disabling ASM didn't help. Still getting the drops.

**Comment by dagg929** [comment]
Did you notice a hike in bandwidth usage just prior to this? Wondering if they just screwed up and when this is hanging, it is actually trying to utilize a ridiculous shitton of down/upload that is causing these hangs.

**Comment by LouDnl** [comment]
Not really noticed it, but could keep an eye out for it.

**Comment by dagg929** [comment]
We had this happen I swear a few months back. I had to down my slots to like 56 versus 64 because the server was using a ton of bandwidth whereas it hadn't before. It would hit a certain point, then we'd get hangs like we are getting now-- only now it doesn't matter if it is a small playercount or a large one.

**Comment by LouDnl** [comment]
Yea I had that too. That's this issue:
http://feedback.arma3.com/view.php?id=23593

**Comment by AgentRev** [comment]
lol BIS denying any involvement and blaming mods, classic

**Comment by LouDnl** [comment]
Yea as usual...

**Comment by hobart666** [comment]
Any lawyers in the Community? We should go after them class-action style... Seriously.

**Comment by dagg929** [comment]
It's at least acknowledged now, huzzah!

**Comment by AgentRev** [comment]
From past experience, acknowledged = we don't give a crap, go away

**Comment by LouDnl** [comment]
We've been monitoring Wasteland and Exile (GamersInc). As we can see this issue does not occur on Exile. It does however occur on Wasteland. But when the server reaches 55+ (or something) players it doesn't seem to happen. Occurrences seem to be random. I think however it is either tie in to players connecting/disconnecting or to the mission system. But no hard proof there. 
We (Creampie and I) think that it might be some decrepated function which is still used and is acting up. But which? No frigging idea. It's very annoying.

**Comment by LouDnl** [comment]
@dagg929 you wrote earlier that you tried default mission. As in Rev's Vanilla A3Wasteland?

**Comment by dagg929** [comment]
I did, yes. With ExtDB2. We had it reoccur with it around 30 people.

My mission file isn't far off from being default now, though I've been tweaking small things on it such as missions or the like.

I had this occur last night on my box with my server at 63, which then killed it down to 5 people by the time it stopped. (BE timed them all out.) I rebooted it at that point and it came up, was acting funny for a few seconds inititially, then stopped. I'm wondering if this causes a huge net spike and that is what is going on when we see this occur, with it then catching up with itself. Has anyone checked their bandwidth usage during this?

**Comment by LouDnl** [comment]
Theres a new perf out: https://www.dropbox.com/sh/582opsto4mmr8d8/3BSy9PdRGm

**Comment by LouDnl** [comment]
Could this be a thing @AgentRev ?
https://community.bistudio.com/wiki/BIS_fnc_MP

From version 1.50 the function bis_fnc_mp will use the engine based remote execution.
This will positively affect its performance, namely processing speed and amount of transferred data.
The syntax and functionality of bis_fnc_mp will remain same to retain the backward compatibility.
To prevent abusing of the remoteExec and remoteExecCall commands the bis_fnc_mp will use, the content authors will be able to define the operation modes as well as white-lists for commands/functions for clients and server separately.
White-lists defined in CfgRemoteExecCommands and CfgRemoteExecFunctions won't be supported from 1.48 as they will be replaced by the new structure more powerful and detailed structure: CfgRemoteExec.

and

Pay close attention to where and when you call BIS_fnc_MP. If you use it on the server during init, it will probably run before the player is loaded into the server, and the player behaves more like a JIP. Either waiting for the briefing to end (time > 0), or using the JIP parameter in BIS_fnc_MP are two ways of getting around this issue.

**Comment by LouDnl** [comment]
Or maybe it's this: https://community.bistudio.com/wiki/BIS_fnc_MPexec

Which is used in filterExecAttempt.sqf ?

**Comment by LouDnl** [comment]
These files use either BIS_fnc_MP or BIS_fnc_MPexec

initPlayerLocal.sqf > vanilla mission
initServer.sqf > vanilla mission
3Dmarkers.sqf > non vanilla addon
APOC_cli_startAirdrop.sqf > non vanilla addon
va_functions.sqf > non vanilla addon
filterExecAttempt.sqf > vanilla mission
fn_vehicleInit.sqf > vanilla mission
fn_MPexec.sqf > vanilla mission
updateConnectingClients.sqf > vanilla mission

Would be nice to know how and to what to replace these functions.

**Comment by LouDnl** [comment]
More information here: https://community.bistudio.com/wiki/Arma_3_Remote_Execution

**Comment by dagg929** [comment]
Anyone get this yet with the newest perf branch?
Comment from Dwarden on the newest: + more crashfixes/freezefixes

Here's hoping.

**Comment by LouDnl** [comment]
Yes still getting them.

**Comment by LouDnl** [comment]
Well I just did a dry test run with a test server.
When first missions start there is a choke. Seems like with every mission start theres some sort of choke.
When I disabled all missions the server runs without chokes for 1.5hours+.

Something within the mission system MUST be choking the server. 
@AgentRev any help here is much appreciated.

**Comment by LouDnl** [comment]
Ofcourse it's always a mix of things. But some chokes definatly come from the mission system. Well atleast it looks like it does. No frigging idea why though.

**Comment by LouDnl** [comment]
Ok so I disabled all addons I could find in (client/server)/init.sqf including r3f, far revive, playericons etc etc. Disabled territories, disabled server cleanup.
Disabled all missions except 1, MBT.

What I see is when the starts up and the mission controllers load theres a micro choke (first red dot) of about a second. Then as soon as the MBT mission starts the choke happens (red line).
http://i.imgur.com/lQ1QadV.png

It seems there is something buried in the missioncontroller/missions that chokes the server.

**Comment by dagg929** [comment]
Is this mission specific? Do you get the same with another mission?

**Comment by LouDnl** [comment]
Also getting this with other missions yes.

**Comment by dagg929** [comment]
So the thought here is that, with players, the server is hanging because of a mission call referencing a bad line of code somewhere-- or a depreciated call. 

Going through my logs now to see if my freezes match the time in my RPT that it is attempting to spawn a mission.

**Comment by dagg929** [comment]
Not able to see the relation on my side but I'll document the freezes more closely and be on the look out now for this.

**Comment by LouDnl** [comment]
When I use ASM to check the freeze time and cross reference that with the rpt log its in 7 of the 10 counts at the exact same time a mission starts.
But theres also 3 times its not. So it's really strange

**Comment by LouDnl** [comment]
The strange thing is, the more players on the server, and thus lower the fps the less frequent the problem.

**Comment by AgentRev** [comment]
If you can provide the RPT log with the time at which the freeze occurs, I can investigate.

**Comment by LouDnl** [comment]
Sure i'll get you one. But there's nothing much to see though.

**Comment by LouDnl** [comment]
Here's a log. Theres nothing to see there though :( I disabled all addons. The choke happened right at the mission start. Only 1 mission in the array.
https://drive.google.com/file/d/0By5PInQAIAA6QVNPYm1MY3FLeW8/view?usp=sharing

**Comment by AgentRev** [comment]
What's the timestamp of the freeze?

**Comment by LouDnl** [comment]
Right on or right before 21:12:54. You can see the mission loading from disk in ASM and then it chokes. And after about 10 seconds it comes back.

**Comment by LouDnl** [comment]
Weird thing is though, with less players on it seems to happen more often then with more players on. I've even seen chokes of 30 minutes.

**Comment by AgentRev** [comment]
> 21:12:54

RPT starts at 22:00:42

**Comment by LouDnl** [comment]
wait huh

**Comment by LouDnl** [comment]
Fuck me wrong link: https://drive.google.com/file/d/0By5PInQAIAA6ZXNEOUxGa3IxRXc/view?usp=sharing

**Comment by LouDnl** [comment]
That log I linked before was for Dwarden. Had chokes too, but don't know the timestamps on that one.

**Comment by LouDnl** [comment]
Ill upload a log of the running server too. Got some timestamps on that one also

**Comment by LouDnl** [comment]
https://drive.google.com/file/d/0By5PInQAIAA6Tk5feHpsNEVjcnM/view?usp=sharing
Timestamps: 
18:50:52
19:31:15 
19:32:52
20:03:37

**Comment by LouDnl** [comment]
Theres more timestamps, but cant get them all

**Comment by AgentRev** [comment]
Well, can't reproduce on my own comp, no chokes at all...

**Comment by LouDnl** [comment]
gah that sucks

**Comment by LouDnl** [comment]
Will try with vanilla again then

**Comment by AgentRev** [comment]
Do the chokes happen even on an empty server?

**Comment by thegoatman** [comment]
For me they don't happen on an empty server

**Comment by LouDnl** [comment]
Yes the logs I gave you were from an empty server.

**Comment by LouDnl** [comment]
Except the last log, this was from our current runnign server.

**Comment by thegoatman** [comment]
@AgentRev if you need it i can give you temp RDP access to a running server if that helps any

**Comment by LouDnl** [comment]
Really strange. Running Vanilla wasteland now and no choke on missions start... This is bugging me out.
Will have to check all updates we did then.

**Comment by AgentRev** [comment]
The only command I know that can wilfully cause such a choke is `nearestObjects` with a high radius

**Comment by thegoatman** [comment]
We don't have something running with nearestobjects that is over 100 meter.

**Comment by AgentRev** [comment]
Then again, if it was fine before v1.52, it's most likely not the issue.

**Comment by AgentRev** [comment]
I'm gonna try running LouD's mission to see if I can reproduce the choke.

**Comment by AgentRev** [comment]
Ran for an hour twice, no chokes.

**Comment by dagg929** [comment]
I cannot reproduce this without players. Rev, did you get at least like 20-30?

We only seem to get this once a night and our server is at 64/64 most of the time. It's very spread out.

**Comment by AgentRev** [comment]
I don't have "players", there's just me. I don't host servers.

**Comment by LouDnl** [comment]
I will do some more runs with different missions today to see if I can reproduce.

**Comment by LouDnl** [comment]
I'm kinda lost here... Unable to reproduce what I could yesterday 7 out of 10 times.

**Comment by LouDnl** [comment]
This the exact log part of a 9 minute choke. You can see the time jump just after the mission starts.
Then after the errors you can see the territory system took 593 seconds.
https://gist.github.com/LouDnl/8636774b0be91baaba3d

**Comment by thegoatman** [comment]
Could it be extdb freezing up?

**Comment by LouDnl** [comment]
Well at this point I don't know what it could be. I really thought it had to do with the mission system because we get drops at the exact same time a mission starts. But I am now unable to reproduce it.
You can see in the log that extDB has sleeps now and then, this wasn't there before 1.52. Already tried updating to the lastest version but that didn't help.
But I can't really tie it to extDB.

**Comment by thegoatman** [comment]
I just had a freeze at the moment someone bought a vehicle.

**Comment by thegoatman** [comment]
Running a test now with inidb to see if the issue remains

**Comment by thegoatman** [comment]
And again a freeze (20 seconds) on the extDB server when a vehicle is bought. So far not a single freeze on the inidb server while stressing it

**Comment by LouDnl** [comment]
That's really weird. You get it on vehicle bought, I get it on mission start. Had reports it also happened when @osuapoc airdrop script was called. 
Maybe it does have to do with this new remoteexec or something? I really don't know and am just guessing here.

**Comment by thegoatman** [comment]
Well if you think about it doesn't the airdrop create a vehicle and stores it with extdb. Missions might be a coincidence with the saving interval?

**Comment by thegoatman** [comment]
inidb still nog showing a freeze when buying insane amouts of objects / starting of missions at 1 second  intervals

**Comment by LouDnl** [comment]
Exile uses extDB2 too and doesn't show these chokes.

**Comment by thegoatman** [comment]
@LouDnl Are you using rev's last dev update with the new join / split string update?

**Comment by LouDnl** [comment]
Yes I was on my own servers. It's not in yet on GamersInc servers. So that's not it.

**Comment by thegoatman** [comment]
@LouDnl Would it be possible for you to add a diag_log on every command when the extdb saving / loading gets called? Am not at my dev machine so can't do it my self. If its not the extdb that will rule it out for sure.

**Comment by LouDnl** [comment]
not atm no sorry. am at work

**Comment by mawendt** [comment]
I'm thinking its tied intowhen the player connect/disconnects and the save equipment file is accessed. I played around a bit on my albeit ancient computer, and found a lag spike exchanging that information... and smooth sailing whenever the save scheme wasn't occurring.

It may be that not only this happens, but chokes noticeably when multiple people on connecting/disconnecting on a server - all trying to write at the same time. But I have seen the action, and it always annoyed me.

And it might be mitigated by having separate extdb's for red/blue/indy/all else. has anyone compared whether the problem occurs when the file gets over a certain size?

**Comment by thegoatman** [comment]
Our dev server was empty and it still had freezes when running with extdb

**Comment by LouDnl** [comment]
Yesterday I could recreate the freezes on an empty server. Today same setup, unable to recreate.

**Comment by thegoatman** [comment]
@LouDnl Whats your saving interval at in the main_config.sqf? Cause on our dev server i had it at 1*60 (default) and I noticed that will make the object saving just run in an endless loop:

11:14:42 "FREEZE CHECK : saveObject.sqf end"
11:14:42 "FREEZE CHECK : saveObject.sqf start"
11:14:42 "FREEZE CHECK : saveObject.sqf end"
11:14:43 "FREEZE CHECK : saveObject.sqf start"
11:14:43 "FREEZE CHECK : saveObject.sqf end"
11:14:43 "FREEZE CHECK : saveObject.sqf start"
11:14:43 "FREEZE CHECK : saveObject.sqf end"
11:14:44 "FREEZE CHECK : saveObject.sqf start"
11:14:44 "FREEZE CHECK : saveObject.sqf end"
11:14:44 "FREEZE CHECK : saveObject.sqf start"
11:14:44 "FREEZE CHECK : saveObject.sqf end"
11:14:44 "FREEZE CHECK : saveObject.sqf start"
11:14:44 "FREEZE CHECK : saveObject.sqf end"
11:14:45 "FREEZE CHECK : saveObject.sqf start"
11:14:45 "FREEZE CHECK : saveObject.sqf end"
11:14:45 "FREEZE CHECK : saveObject.sqf start"
11:14:45 "FREEZE CHECK : saveObject.sqf end"
11:14:46 "FREEZE CHECK : saveObject.sqf start"
11:14:46 "FREEZE CHECK : saveObject.sqf end"
11:14:46 "FREEZE CHECK : saveObject.sqf start"
11:14:46 "FREEZE CHECK : saveObject.sqf end"
11:14:46 "FREEZE CHECK : saveObject.sqf start"
11:14:46 "FREEZE CHECK : saveObject.sqf end"
11:14:47 "FREEZE CHECK : saveObject.sqf start"
11:14:47 "FREEZE CHECK : saveObject.sqf end"

**Comment by LouDnl** [comment]
I run the saving on a headlessclient an that doesn't frees. The main server still freezes.
I still see the freezes around mission starts.

**Comment by speedweasel** [comment]
Can you do some broad testing by disabling entire systems:

The Territory system
Missions
Anti-hack
Persistence

**Comment by speedweasel** [comment]
Can you do some broad testing by disabling entire systems?

The Territory system
Missions
Anti-hack
Persistence
Loot spawner
R3F Logistics
etc...

Maybe it can be narrowed down to a specific system/addon.

**Comment by LouDnl** [comment]
I'm seeing no chokes on my test server atm. On our new host it's just random.

**Comment by LouDnl** [comment]
I disabled the airdrop system by Apoc as it seems it causes chokes and lags too.

**Comment by thegoatman** [comment]
Hmm then am starting to think. Can it be createVehicle causing freezes? Sinds you have it on mission starts and I had it on vehiclestore buying

**Comment by LouDnl** [comment]
Just a wild guess. But all vehicle spawns, buying, mission and Apoc airdrop call vehicleSetup. Maybe it's something in that file? Just guessing here as i'm kinda lost.

**Comment by Bl4ckcircle** [comment]
Someone try with inidb ?

**Comment by hobart666** [comment]
LouD I disabled Airdrop sys about a day after the update it helped a little bit but no matter how many ppl are in me server the lag experienced in there is completely random when it happens and only lasts for 10-30 sec tops. I shut down my HC and got more complaints (Mainly for the vehicle saving issues I mentioned to you & rev a month or so ago, with HC going no one gets those anymore oddly) after a few days I started running it again and most have subsided except the random lag from BIS ( I blame them, the mission/server was fine before their BS update as usual.) , Here's a fun fact, I also run HL based servers on the same box and since the last update every time someone connects to one of my other servers ppl in the arma server start getting the "losing connection" hint, and sometimes they get dropped. I have no lag or frame drops or anything on my other servers both running very nicely @ 128 ticks with 0 connection issues.

**Comment by hobart666** [comment]
3:14:38 PM | RCon admin #0: (Global) BEC: zy you frozen?
3:14:51 PM | (Global) ZyklonBeast: yeah
3:14:52 PM | (Global) CallousDisregard: can't let go of my wall
3:14:55 PM | (Global) ZyklonBeast: that was a long one
3:15:04 PM | (Global) CallousDisregard: nvm ...it's gone
3:15:16 PM | RCon admin #0: (Global) BEC: is it a lag spike thing?
3:15:19 PM | (Global) ZyklonBeast: yeah
3:15:46 PM | (Global) CallousDisregard: seems like the datbase locks up and you lose what yo uare holding when it catches up
3:16:18 PM | RCon admin #1: (Global) BEC : Server will restart in 5 minutes
3:16:21 PM | RCon admin #0: (Global) BEC: do these happen around any events such as someone purchasing a vehicle from the store or a mission starting?
3:16:26 PM | (Global) CallousDisregard: you can order several veh and they all arrive at once, veh are rearming ..i have a UAV showing 40 reloads when it should have none
3:16:31 PM | (Global) ZyklonBeast: trying to lock onto a butterfly
3:16:50 PM | (Global) CallousDisregard: they happen at complete random, even when only 1 person on the server
3:16:56 PM | RCon admin #0: (Global) BEC: anything involving a vehicle being spawned?
3:16:59 PM | (Global) CallousDisregard: sometimes AI freeze in place
3:17:01 PM | RCon admin #0: (Global) BEC: well fuck
3:17:10 PM | RCon admin #0: (Global) BEC: jeebus this shit
3:17:22 PM | (Global) ZyklonBeast: you can tell if its server lag if the AI and missions arent moving
3:17:26 PM | (Global) CallousDisregard: the stores will reject your choice then later spawn it in

this is just a few mins ago before restart all time-stamps are Eastern Standard Time USA

**Comment by thegoatman** [comment]
This might be related to it. I noticed all armored vehicles are getting extra magazines after a restart, i.e. we got a blackfoot that has multiple flare and ammo mags where before 1.52 it didn't. And a couple of tank got 32 magazines for the commander gun.

**Comment by LouDnl** [comment]
Hmmm I think it's all vehicles. You can hear multiple reload sounds even when you buy a vehicle. Hence the choke also happens on an airdrop?

**Comment by osuapoc** [comment]
Check the vehicle setup function for magazine addition.

-----Original Message-----

**Comment by LouDnl** [comment]
No weird things there. Only thing I can see is minigun stuff:
https://github.com/A3Wasteland/ArmA3_Wasteland.Altis/blob/master/server/functions/vehicleSetup.sqf#L128-L138

**Comment by thegoatman** [comment]
It might be in vLoad.sqf? sinds that is handeling magazines and reloading the vehicles

**Comment by LouDnl** [comment]
That would only count for vehicles on server start. As these chokes are happening during complete uptime it can't be limited to loaded in vehicles.

**Comment by hobart666** [comment]
Ok, so randomly after a couple days I am having 0 issues running smoothly for 6 hours ( with missions & extra random ai script  as well) @ 49.8 server frames and no lag spiking or freezing at all. secret bohimia fix?

**Comment by LouDnl** [comment]
Nope still having them.

**Comment by hobart666** [comment]
check mine out I don't know what I did ( I did change a base name ) in the mission.sqm. Yes I'm still using that! 162.248.95.195:2302

**Comment by LouDnl** [comment]
Chokes are usually logged like this:

11:54:25 NetServer::finishDestroyPlayer(432552428): DESTROY immediately after CREATE, both cancelled
11:55:03 NetServer::finishDestroyPlayer(498051416): DESTROY immediately after CREATE, both cancelled
11:55:29 Error: Object(19 : 26) not found
11:55:29 Error: Object(19 : 26) not found
11:55:29 Error: Object(19 : 27) not found
11:55:29 Error: Object(19 : 27) not found
11:55:29 Error: Object(18 : 30) not found
11:55:29 Error: Object(18 : 30) not found
11:55:29 Error: Object(10 : 89) not found
11:55:29 Error: Object(10 : 89) not found
11:55:30 R Alpha 2-1:1 (Tony Ponti) REMOTE: Getting out while IsMoveOutInProgress
11:55:33 NetServer::SendMsg: cannot find channel #539195539, users.card=8
11:55:33 NetServer: users.get failed when sending to 539195539
11:55:33 Message not sent - error 0, message ID = ffffffff, to 539195539 (<no>)
11:55:33 NetServer::SendMsg: cannot find channel #539195539, users.card=8
11:55:33 NetServer: users.get failed when sending to 539195539
11:55:33 Message not sent - error 0, message ID = ffffffff, to 539195539 (<no>)
11:55:33 NetServer::SendMsg: cannot find channel #539195539, users.card=8
11:55:33 NetServer: users.get failed when sending to 539195539
11:55:34 Server: Network message 4c62a4 is pending
11:55:34 Server: Network message 4c62a6 is pending
11:55:34 Server: Network message 4c62a6 is pending
11:55:34 Server: Network message 4c62a6 is pending
11:55:34 Server: Network message 4c62a8 is pending
11:55:34 Server: Network message 4c62a8 is pending

**Comment by LouDnl** [comment]
@l0dac are the TOP servers having this problem?

**Comment by LouDnl** [comment]
Seems like @A3Armory / @Gigatek1 are also experiencing the same problems and they are on CouchDB. So this rules out extDB2

**Comment by LouDnl** [comment]
I think the chokes are tied in to this: https://github.com/A3Wasteland/ArmA3_Wasteland.Altis/issues/356

And the lag is seriously tied in to update 1.52.

**Comment by hobart666** [comment]
The lag went away for me when I deleted all vehicles. It only returned after patrons started saving them again and after a few restarts when the turret ammo was being duped. Start with fresh vehicles and over the first few restarts the the server runs PERFECT. Then drops into the abyss.

**Comment by LouDnl** [comment]
Seems that the chokes are gone here now that I disabled turret ammo saving and loading.

**Comment by hobart666** [comment]
Not here totally, but it is a ton better, they've decreased in time significantly instead of a spike lasting a minute or two it's only for a few seconds now.

**Comment by LouDnl** [comment]
Haven't seem them yet.

**Comment by LouDnl** [comment]
Running on perf6 now, still seeing some server lag. No chokes since I disabled the turret saving/loading yet.

**Comment by LouDnl** [comment]
Scrap my last comment. Just had a choke.

**Comment by hobart666** [comment]
rgr tht

**Comment by hobart666** [comment]
looking for commas I added by mistake, I know I put one here somewhere!

**Comment by Gigatek1** [comment]
Is anyone still getting the lag and desync chokes? We switched to v8 of the perf binaries which pretty much fixed it. When we switched to v9 the desync and lag came back.  So the fix for us was somewhere in v8.

**Comment by thegoatman** [comment]
Not had a single freeze up after removing the saving of all turret ammo. Running on perf 7.

**Comment by LouDnl** [comment]
V9 brought back chokes yes.

**Comment by Gigatek1** [comment]
I talked to some other players and it seems v8 wasn't a complete fix. It just didn't happen as bad with v7 or v9. I'm going to try something for a fix and see how it goes tomorrow (Sunday). I'm going to put vehicle saving to more then 6 hours (we do restarts every 6 hours) to prevent saving of vehicles and only have vehicles saved during restarts. The version we use has a sock script that kicks everyone from the server then saves everything before the server shut down. Hopefully we don't have any crashes. 

I was told that the chokes were happening every 20 minutes. That's what we have saving set to.
https://github.com/A3Armory/ArmA3_Wasteland.Altis/blob/dev/server/default_config.sqf#L81

**Comment by hobart666** [comment]
I am as of yesterday and during the spike this is flooded in my .rpt
Server: Network message 3dc9a7 is pending

and this was in there too just before those other lines.

"onPlayerDisconnected - ["Quin","76561198061494241"]"
Client: Remote object 16:0 not found
"HandleDisconnect - ["Quin","76561198061494241"] - alive: true - local: true"
Warning: Cleanup player - person 2:10888 not found
Warning: Cleanup player - person 2:10888 not found
"TERRITORY SYSTEM: _realLoopTime was 10.4863"
"TERRITORY SYSTEM: _realLoopTime was 10.4658"
"WASTELAND SERVER - Money Mission 2 started: Military Patrol"
"TERRITORY SYSTEM: _realLoopTime was 10.4043"
Performance warning: SimpleSerialization::Write 'messageSystem' is using type of ',TEXT' which is not optimized by simple serialization, falling back to generic serialization, use generic type or ask for optimizations for these types
"WASTELAND SERVER - Money Mission 2 waiting to be finished: Military Patrol"
Performance warning: SimpleSerialization::Read 'messageSystem' is using type of ,'TEXT' which is not optimized by simple serialization, falling back to generic serialization, use generic type or ask for optimizations for these types
"TERRITORY SYSTEM: _realLoopTime was 10.4375"
"TERRITORY SYSTEM: _realLoopTime was 10.3135"
"TERRITORY SYSTEM: _realLoopTime was 10.3799"
"TERRITORY SYSTEM: _realLoopTime was 10.4326"
"TERRITORY SYSTEM: _realLoopTime was 10.4131"
"TERRITORY SYSTEM: _realLoopTime was 10.458"
"TERRITORY SYSTEM: _realLoopTime was 10.4492"
Error: Object(4 : 130) not found
Error: Object(4 : 130) not found
Server: Network message 3dc9a7 is pending
Server: Network message 3dc9a7 is pending
